packages/dataviews/src/dataform-controls/date.tsx

```diff
@@ -520,6 +714,9 @@ export default function DateControl< Item >( {
 			onChange={ onChangeCalendarDateControl }
 			label={ label }
 			hideLabelFromVision={ hideLabelFromVision }
+			data={ data }
+			field={ field }
+			setValue={ setValue }
```

Comment:
This is a bit too much. For example, `setValue` is already available via the field, and the `value` can be calculated via `field.getValue` as well. Can we pass to this component (and the others) the same props as `DataFormControlProps`?